enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+     * remoteIpAddress: string
+     * remoteMac: string
+     * successful: boolean
+   * l3Connectivity
+     * outgoingNic: string
+     * remoteIpAddress: string
+     * successful: boolean
+ * apiVipConnectivity: boolean
+ * ntpSources: array of
+   * sourceName: string
+   * sourceState: [synced, combined, not_combined, error, variable, unreachable]
+
+
+#### REST API Access
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
+authentication headers, the Assisted Service must generate some random URL or
+query parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+AgentDrivenHostInstallation CRs itself, the agent will continue communicating
+with the service via REST APIs.  Currently the service embeds the user’s pull
+secret in the discovery ISO which the agent passes in an authentication
+header.  In this case the service can generate and embed some token which it
+can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include whatever information is necessary to get an
+ISO and boot hosts.
+
+A new field in the ClusterDeployment spec will optionally reference an
+AgentDrivenClusterInstallation resource, which describes everything that the
+agent-based install tooling needs to know about the cluster that it will be
+creating. Some or all of that new resource may be folded into ClusterDeployment
+in the future, but in the immediate term, keeping it separate will enable much
+quicker development.
+
+Hive or ACM will optionally orchestrate the booting of bare metal hosts by:
+
+1. Get an ISO URL from assisted operator.
+1. Use metal3's BareMetalHost to boot the live ISO on some hardware.
+
+#### BareMetalHost can boot live ISOs
+
+A [separate enhancement to
+metal3](https://github.com/metal3-io/metal3-docs/pull/150) proposes a new
+capability in the BareMetalHost API enabling it to boot live ISOs. That feature
+is required so that automation in a cluster can boot the discovery ISO on known
+hardware as the first step toward provisioning that hardware.
+
+#### CAPBM 
+
+[Cluster-API Provider Bare Metal](https://github.com/openshift/cluster-api-provider-baremetal/) will need to gain the ability to interact with the new assisted installer APIs. Specifically it will need to match a Machine with an available assisted Agent that is ready to be provisioned. This will be in addition to matching a Machine with a BareMetalHost, which it already does today.
```

Comment:
I'd like to see some discussion here of the alternative - a new machine API provider specifically for agent-driven installation, such that agent-driven support isn't limited to bare metal

The basic idea is straightforward - a new actuator that simply satisfies a Machine request by choosing an AgentDrivenHostInstallation resource and initiating an install

The challenging part is how to then leverage the per-platform providers to make it easy to boot a new machine with the agent, in the case where there is no existing AgentDrivenHostInstallation available

Reply:
Another reason why it's important to consider this alternative - we may struggle to find good justification why support for agent-driven installation is within the scope of upstream CAPBM, and maye for this exact reason ... that there is nothing bare-metal specific about it

Reply:
Also, consider the boot-it-yourself day 2 add-worker case. There really is no need for a BareMetalHost resource to be involved with such a workflow

Reply:
@mhrivnak This thread was detached from the text because of changes, but I don't think it's resolved.

Reply:
I think we just agreed to delete this whole section, and the proposal no longer proposes any particular Machine API integration. There's a new section `Future Use of MachinePools and Machine API` that elaborates.